internal/component/prometheus/scrape/scrape.go

```diff
@@ -290,10 +306,12 @@ func New(o component.Options, args Arguments) (*Component, error) {
 
 	alloyAppendable := prometheus.NewFanout(args.ForwardTo, o.ID, o.Registerer, ls)
 	scrapeOptions := &scrape.Options{
+		// NOTE: This is not Update()-able.
```

Comment:
It's very weird to have non-updateable arguments. Does it have to be this way? It'd be worth at least documenting it so that it doesn't trip someone up.